packages/css/table.css

```
@@ -1,143 +1,149 @@
+  --dsc-table-background--hover: var(--ds-color-neutral-surface-default);
+  --dsc-table-background--zebra: var(--ds-color-neutral-background-subtle);
+  --dsc-table-background: var(--ds-color-neutral-background-default);
+  --dsc-table-border-radius: 0;
+  --dsc-table-border: 1px solid var(--ds-color-neutral-border-subtle);
   --dsc-table-color: var(--ds-color-neutral-text-default);
-  --dsc-table-header-cell-background: var(--ds-color-neutral-background-default);
-  --dsc-table-header-sorted-background: var(--ds-color-neutral-background-subtle);
-  --dsc-table-header-sorted-hover: var(--ds-color-neutral-surface-default);
-  --dsc-table-cell-backround: var(--ds-color-neutral-background-default);
-  --dsc-table-cell-zebra-background: var(--ds-color-neutral-background-subtle);
-  --dsc-table-cell-hover-backround: var(--ds-color-neutral-surface-default);
-
-  position: relative;
-  border-collapse: separate;
-  border-spacing: 0;
-  text-align: left;
-  color: var(--dsc-table-color);
-  width: 100%;
-}
-
-.ds-table--sticky-header {
-  overflow: auto;
-}
-
-.ds-table--border .ds-table__row:last-of-type td {
-  border-bottom: 0;
-}
-
-.ds-table--sm {
-  --dsc-table-padding: var(--ds-spacing-1) var(--ds-spacing-3);
-}
-
-.ds-table--md {
+  --dsc-table-header-background--hover: var(--ds-color-neutral-surface-default);
+  --dsc-table-header-background--sorted: var(--ds-color-neutral-background-subtle);
+  --dsc-table-header-background: var(--ds-color-neutral-background-default);
+  --dsc-table-header-divider: 2px solid var(--ds-color-neutral-border-subtle);
   --dsc-table-padding: var(--ds-spacing-2) var(--ds-spacing-3);
-}
+  --dsc-table-sort-size: var(--ds-spacing-6);
 
-.ds-table--lg {
-  --dsc-table-padding: var(--ds-spacing-3) var(--ds-spacing-3);
-}
-
-.ds-table__head {
-  z-index: 0;
-  box-sizing: border-box;
-  font-family: inherit;
+  border-collapse: separate; /* Using separate mode to enable border-radius */
+  border-radius: var(--dsc-table-border-radius);
   border-spacing: 0;
-  border-bottom: 2px solid var(--dsc-table-border-color);
-}
+  color: var(--dsc-table-color);
+  width: 100%;
 
-.ds-table__header__cell {
-  padding: var(--dsc-table-padding);
-  font-family: inherit;
-  background-color: var(--dsc-table-header-cell-background);
-  border-spacing: 0;
-  border-bottom: 2px solid var(--dsc-table-border-color);
-}
+  & > :is(tbody, thead) > tr > :is(th, td) {
+    background: var(--dsc-table-background);
+    border-bottom: var(--dsc-table-border);
+    padding: var(--dsc-table-padding);
+    text-align: inherit;
 
-.ds-table--sticky-header .ds-table__head .ds-table__header__cell {
-  position: sticky;
-  top: 0;
-  z-index: 1;
-}
+    &:is(th) {
+      font-weight: 500;
+    }
+  }
 
-.ds-table__header__cell--sortable {
-  padding: 0;
-}
+  & > thead > tr:last-child > :is(th, td) {
+    border-bottom: var(--dsc-table-header-divider);
+    background: var(--dsc-table-header-background);
+  }
```

```suggestion
  & > thead > tr > :is(th, td) {
    background: var(--dsc-table-header-background)
  }

  & > thead > tr:last-child > :is(th, td) {
    border-bottom: var(--dsc-table-header-divider);
  }
```
We need to make sure bavkground is added to all elements in the header, but border only to the ones inside the last `tr`. The current code would not add a background to the children of the first `tr`

---

Good point! Added a story with multiple header rows so we don't forget this in the future :)